cmd/utils.go

```diff
@@ -126,6 +126,72 @@ func parseEnvURL(envURL string) (*url.URL, string, string, *probe.Error) {
 	return u, accessKey, secretKey, nil
 }
 
+func newS3Config(args cli.Args) (*Config, *probe.Error) {
+	var (
+		url       = args.Get(0)
+		accessKey = args.Get(1)
+		secretKey = args.Get(2)
+		api       = args.Get(3)
+	)
+
+	s3Config := &Config{
+		// Default config parameters
+		AppName:     "mc",
+		AppVersion:  Version,
+		AppComments: []string{os.Args[0], runtime.GOOS, runtime.GOARCH},
+		Debug:       globalDebug,
+		Insecure:    globalInsecure,
+		// S3 connection parameters
+		AccessKey: accessKey,
+		SecretKey: secretKey,
+		HostURL:   url,
+	}
+
+	// If api is provided we do not auto probe signature, this is
+	// required in situations when signature type is provided by the user.
+	if api != "" {
+		s3Config.Signature = api
+		return s3Config, nil
+	}
+
+	// Test s3 connection for API auto probe
+	s3Config.Signature = "s3v4"
+	s3Config.HostURL = urlJoinPath(url, "probe-bucket-sign")
+
+	s3Client, err := s3New(s3Config)
+	if err != nil {
+		return nil, err.Trace(args...)
+	}
+
+	if _, err = s3Client.Stat(false, false); err != nil {
```

Comment:
Can you convert this into like a `probeSignature`  function which returns either "s3v4" or "s3v2" along with error.?